docs/contributing.md

```diff
@@ -0,0 +1,143 @@
+
+Unix/macOS:
+```
+python -m pip install -r requirements.txt
+```
+
+Windows:
+```
+py -m pip install -r requirements.txt
+```
+
+### Conda Setup
+
+These instructions rely on the [conda](https://docs.conda.io/en/latest/) package
+manager. Run the following from the base of the repository to create a new conda
+environment from the `environment.yml` file:
+
+```bash
+conda env create
+```
+
+Before building and testing the project, you have to activate the environment
+(you'll need to do this every time you start a new terminal):
+
+```bash
+conda activate gmt-examples
+```
+
+## Building the documentation
+
+Once you have [setup your environment](#setting-up-your-environment), you can
+build the documentation using:
+
+```bash
+    cd docs
+    make html
+```
+
+## GMT Examples Overview
+
+There are two main components to GMT examples project:
+
+* Gallery examples, with source material in the `docs/examples/gallery/` folder.
+* Tutorial examples, with source material in the `docs/examples/tutorials/` folder.
+
+The gallery examples are designed to instruct users on how to complete a specific
+problem. For general recommendations on how to design effective gallery examples,
+see the [diataxis framework's section on how-to guides](https://diataxis.fr/how-to-guides/).
+
+The tutorials are learning orientated with the goal of teaching users GMT. For
+general recommendations on how to design effective tutorials, see the
+[diataxis framework's section on tutorials](https://diataxis.fr/tutorials/).
+
+The documentation are written primarily in
+[reStructuredText](https://docutils.sourceforge.io/rst.html) and built by
+[Sphinx](http://www.sphinx-doc.org/). Please refer to
+[reStructuredText Cheatsheet](https://docs.generic-mapping-tools.org/latest/devdocs/rst-cheatsheet.html)
+if you are new to reStructuredText.
+
+## Contributing New Examples
+
+The source files for the gallery examples and tutorials are `.rst` files in
+`docs/examples/` that generate one or more figures using the
+[sphinx_gmt](https://www.generic-mapping-tools.org/sphinx_gmt/latest/)
+extension. To add a new gallery example or tutorial:
+
+* If necessary, create a new sub-directory under `docs/examples/gallery/` for a
+  gallery example section (e.g., `basemaps/` or `plot_embellishments/`).
+* Create a new `.rst` file inside the appropriate sub-directory in
+  `docs/examples/gallery/` or `docs/examples/tutorials/`.
+* Add a descriptive title and as much explanation as necessary.
+* Add hyperlinks to GMT modules using the `gmt-module` restructured text directive:
+
+  ```bash
+  :gmt-module:`grdview`
+  ```
+
+* Add hyperlinks to GMT module options by appending lower-case `#<option>` to
+  the `gmt-module` argument:
+
+  ```bash
+  :gmt-module:`grdview#q`
+  ```
+
+* Add as many figures as needed using the `gmtplot` directive:
+
+  ```bash
+  .. gmtplot::
+     :language: bash
```

Comment:
I think most examples (if not all) will in written in Bash. So, it makes sense to set the default language to "bash", so contributors don't have to specify the language option explicitly. The default language is set by the `highlight_language` setting in the conf.py, as mentioned in https://www.generic-mapping-tools.org/sphinx_gmt/latest/api/generated/sphinx_gmt.gmtplot.html#module-sphinx_gmt.gmtplot.